### DISCUSSION TITLE
Slowed down hiding of checked off habit in 2.1.1

### DISCUSSION THREAD

**Discussion by aubreyz** [iSoron/uhabits#1550] [Technical Discussion] [answered]
Thanks as always for the fantastic app. 

In 2.1.1 I noticed that when "Hide completed" is enabled, it takes much longer than before (too long) for the line to disappear when checked. Before it was instant - now it takes about 2 seconds which is too long. The problem is that if you are checking off more than one habit, then the things are moving around when checking off the second one, leading to errors. 

Not a big problem, but just pointing this out in case it is deliberate, or a side effect of something else, or if anyone else has noticed this. 

## System information
 - Phone: Moto G7
 - Phone Operating System: Android 10
 - App version: 2.1.1

**Comment by iSoron** [comment]
Thank you for the bug report, @aubreyz. The delay is deliberate; see #1055.

**Comment by aubreyz** [comment]
Ok but this is a functionality issue that makes the app worse. Is there a way to switch this off??  It is wildly too long leading to problems of incorrect checking off of tasks as they jump around at future points in time. ☹️

**Comment by hiqua** [comment]
@iSoron it could make sense to remove the delay when the user is using the pop-up. It was initially introduced so that the user could cycle through the states using multiple presses, but if you use the pop-up selector you will only press once anyway. What do you think?

**Comment by iSoron** [comment]
@hiqua Yes, that makes sense.

**Reply by hiqua** [reply]
@iSoron Actually I'm thinking of completely removing the delay I introduced in https://github.com/iSoron/uhabits/pull/1147: if a user often wants to enter a state different than just ticking, they should disable the short press option and use the pop-up instead.

**Reply by iSoron** [reply]
@hiqua I think the delay is a good idea, so I am not sure about completely removing it, but I agree that it currently has issues. Perhaps it could work if we (i) made the delay reset global, instead of per-habit; (ii) removed it from checkmarks entered using the popup.

**Comment by t214c** [comment]
@iSoron @hiqua If so, please show a toast as confirmation e.g. "Habit X marked ✔"

**Comment by aubreyz** [comment]
I am probably being a bit dense here, but I am not sure what popup is being referred to. The only time I see any sort of popup is with numeric habits. Where is a pop up? 

What I see is rows of horizontal habits, and with ""hide completed" checked. I I want to mark just one habit as done that is fine.
But what generally happens is that I would do three or four things at a time and then quickly tick, tick, tick them all at once (I often color them the same and then sort by color so they appear one after the other

Brush Teeth
Wash face
Polish Nose
Trim nose hair

The problem with the large delay, is that the rows are moving around as they disappear much more slowly than I would be ticking them. By the time my finget hits the tick mark a completely different habit has moved to under the finger. 

It is probably because I am not seeing (or using) this popup thing that I don't understand -- no problem with a toast if it does not get in the way, but if a line is erased after poking it with the finger surely that tells you it has been marked as done?

**Comment by hiqua** [comment]
> Where is a pop up?

If you disable "Toggle with short press", then a short press should open some pop-up with the option for the entry:
* yes
* no
* non-entered (question mark)
And the "notes" field which is free form.

Ideally we'd also avoid sorting if a long-press is being done.

**Comment by aubreyz** [comment, marked as answer]
Here is a hint for people who are about to give up on this app due to this problem.  You MUST always check off habits going from the bottom of the screen upwards, and never from top to bottom, otherwise you will never get around the problem of clicking on the wrong entry by mistake all the time. 

I really don't know why this massive bug was converted into a "discussion". It is a massive and clear user interface problem. I really must beg that this is fixed. 

I used to love this application -- and I feel so badly that I am about to give up on it in sheer frustration. Why on earth was this very bad change introduced???
